modules/distr-tracing-tempo-object-storage-setup-azure-sts-install.adoc

```diff
@@ -0,0 +1,126 @@
+  --audiences <audience> # <5>
+----
+<1> Federated identity credentials allow Kubernetes service accounts to authenticate as an Azure managed identity without storing secrets or using an Azure service principal identity.
+<2> Choose a name for the federated credential.
+<3> The URL of the OpenID Connect (OIDC) issuer for your cluster.
+<4> The service account subject for your cluster in the following format: `system:serviceaccount:<namespace>:tempo-<tempostack_instance_name>`.
+<5> The expected audience, which is to be used for validating the issued tokens for the federated identity credential. This is commonly set to `api://AzureADTokenExchange`.
++
+[TIP]
+====
+You can get the URL of the OpenID Connect (OIDC) issuer for your cluster by running the following command:
+----
+$ oc get authentication cluster -o json | jq -r .spec.serviceAccountIssuer
+----
+====
+
+. Create a federated identity credential for the Kubernetes service account for use by the frontend component:
++
+[source,terminal]
+----
+$ az identity federated-credential create \ # <1>
+  --name <credential_name>-frontend \ # <2>
+  --identity-name <identity_name> \
+  --resource-group <resource_group> \
+  --issuer <cluster_issuer> \
+  --subject <tempo_sa_query_frontend_subject> \ # <3>
+  --audiences <audience> | jq
+----
+<1> Federated identity credentials allow Kubernetes service accounts to authenticate as an Azure managed identity without storing secrets or using an Azure service principal identity.
+<2> Choose a name for the frontend federated identity credential.
+<3> The service account subject for your cluster in the following format: `system:serviceaccount:<namespace>:tempo-<tempostack_instance_name>`.
+
+. Assign the Storage Blob Data Contributor role to the Azure service principal identity of your created Azure managed identity:
++
+[source,terminal]
+----
+$ az role assignment create \
+  --assignee <assignee_name> \ # <1>
+  --role "Storage Blob Data Contributor" \
+  --scope "/subscriptions/<subscription_id>
+----
+<1> The Azure service principal identity of the Azure managed identity that you created in step 1.
++
+[TIP]
+====
+You can get the `<assignee_name>` value by running the following command:
+----
+$ az ad sp list --all --filter "servicePrincipalType eq 'ManagedIdentity'" | jq -r --arg idName <identity_name> '.[] | select(.displayName == $idName) | .appId'`
+----
+====
+
+. Fetch the client ID of the Azure managed identity that you created in step 1:
++
+[source,bash]
+----
+CLIENT_ID=$(az identity show \
+  --name <identity_name> \ # <1>
+  --resource-group <resource_group> \ # <2>
+  --query clientId \
+  -o tsv)
+----
+<1> Copy and paste the `<identity_name>` value from step 1.
+<2> Copy and paste the `<resource_group>` value from step 1.
+
+. Create a Kubernetes secret for the Azure workload identity federation (WIF):
++
+[source,terminal]
+----
+$ kubectl create -n <tempo_namespace> secret generic azure-secret \
```

Comment:
I think we will need an `oc` command here. Can you please check if you can provide an `oc` command for this block?

Reply:
I think will be the same, just replace kubectl with oc.

Reply:
Thank you. I'll update and then ask Ishwar to check this.